bundle/EventListener/ScheduledVisibilityListener.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\EventListener;
+
+use Ibexa\Contracts\Core\Repository\Events\Content\PublishVersionEvent;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Configuration\ScheduledVisibilityConfiguration;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Enums\VisibilityUpdateResult;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Exception\InvalidStateException;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Service\ScheduledVisibilityService;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+use function sprintf;
+
+final class ScheduledVisibilityListener implements EventSubscriberInterface
+{
+    public function __construct(
+        private readonly ScheduledVisibilityService $scheduledVisibilityService,
+        private readonly ScheduledVisibilityConfiguration $configurationService,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    ) {}
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            PublishVersionEvent::class => ['onPublishVersion'],
+        ];
+    }
+
+    public function onPublishVersion(PublishVersionEvent $event): void
+    {
+        $content = $event->getContent();
+
+        if (!$this->configurationService->isEnabled()) {
```

Comment:
Move applicability checks to the very top.

Reply:
fixed in 82511aefc7903156c1f57730f612de20f2cbc1de